### PR TITLE
C API for Visualization: Round 2

### PR DESCRIPTION
**Pull request by znation**
With this commit, C API for visualization is at feature parity with
the internal-facing C++ API.

This change adds:

* Complete support for 2d and SFrame summary plots
* Proper streaming support
* Complete unit testing across all plot types

### CODE REVIEW

File: src/unity/lib/visualization/plot.hpp [hunk 1]
```diff
@@ -26,7 +26,8 @@ namespace turi {
         std::string get_spec();
 
         // streaming data aggregation
-        bool finished();
+        double get_percent_complete(); // out of 1.0
```
Comment (TobyRoseman) [inline src/unity/lib/visualization/plot.hpp]: Could these two methods be const methods?

File: src/capi/impl/capi_visualization.cpp [hunk 2]
```diff
@@ -23,13 +24,59 @@ extern "C" {
...
+        turi::ensure_server_initialized();
+
+        CHECK_NOT_NULL(error, sa_x, "sarray_x", NULL);
+        CHECK_NOT_NULL(error, sa_y, "sarray_y", NULL);
+        CHECK_NOT_NULL(error, title, "title", NULL);
+        CHECK_NOT_NULL(error, x_axis_title, "x_axis_title", NULL);
+        CHECK_NOT_NULL(error, y_axis_title, "y_axis_title", NULL);
+
+        std::shared_ptr<turi::model_base> plot = turi::visualization::plot(
+            "" /* path_to_client */,
```
Comment (TobyRoseman) [inline src/capi/impl/capi_visualization.cpp]: Why is this the empty string?

**Comment by znation** [comment]
@TobyRoseman Feedback addressed; please take a look again, thanks!

**Comment by TobyRoseman** [comment]
@znation - LGTM

**Pull request merged**